service/src/main/java/org/cbioportal/service/ClinicalDataService.java

```diff
@@ -52,9 +54,7 @@ BaseMeta fetchMetaClinicalData(List<String> studyIds, List<String> ids, List<Str
     List<ClinicalData> getPatientClinicalDataDetailedToSample(List<String> studyIds, List<String> patientIds,
             List<String> attributeIds);
 
-    List<ClinicalData> fetchSampleClinicalTable(List<String> studyIds, List<String> sampleIds, Integer pageSize,
-                                                Integer pageNumber, String searchTerm, String sortBy, String direction);
-    
-    Integer fetchSampleClinicalTableCount(List<String> studyIds, List<String> sampleIds, String searchTerm,
-                                          String sortBy, String direction);
+    ImmutablePair<SampleClinicalDataCollection, Integer> fetchSampleClinicalTable(List<String> studyIds, List<String> sampleIds, Integer pageSize,
```

Comment:
It's great that we consolidate these two functions into one.

Reply:
Could we add a comment on what the two elements in the pair are, I think people can't derive the meaning from the naming by themselves. Or maybe we can rename this function?

Reply:
I have created a wrapper class that serves as documentation:
```
public class ClinicalDataTableResponse {
    
    private final ImmutablePair<SampleClinicalDataCollection, Integer> pair;
    
    public ClinicalDataTableResponse (SampleClinicalDataCollection pagedCollectionFragment, Integer totalCount) {
        this.pair = new ImmutablePair<>(pagedCollectionFragment, totalCount);
    }

    public ClinicalDataTableResponse() {
        this.pair =  new ImmutablePair<>(new SampleClinicalDataCollection(), 0);
    }
    
    public SampleClinicalDataCollection getSampleClinicalDataCollection() {
        return pair.getLeft();
    }
    public Integer getTotalCount() {
        return pair.getRight();
    }
    
}
```

What do you think of this?

Reply:
This looks good, thanks!